packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -231,6 +234,295 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
+
+  for (let i = 0; i < 30; i++) {
+    axios
+      .get(domain)
+      .then(() => {
+        log("gateway is reachable");
+        reachable = true;
+      })
+      .catch(() => {
+        log("gateway is not reachable");
+      });
+    if (reachable) {
+      break;
+    }
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
+  }
+
+  if (reachable) {
+    axios.get(domain).then(res => {
+      log(res.status);
+      log(res.statusText);
+      log(res.data);
+      expect(res.status).toBe(200);
+      expect(res.statusText).toBe("OK");
+      expect(res.data).toContain("Directory listing for /");
+      expect(res.data).toContain("bin/");
+      expect(res.data).toContain("dev/");
+      expect(res.data).toContain("etc/");
+    });
+  } else {
+    throw new Error("Gateway is unreachable after multiple retries");
+  }
+
+  //--------------------IPv6--------------------
+  backends = ["http://[" + result[0].publicIP["ip6"].split("/")[0] + "]:8000"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw2: GatewayNameModel = {
+    name: name + "6",
+    node_id: gatewayNodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  gatewayRes = await gridClient.gateway.deploy_name(gw2);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw2.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "6");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  domain = "https://" + gatewayResult[0].domain;
+  reachable = false;
+
+  for (let i = 0; i < 30; i++) {
+    axios
+      .get(domain)
+      .then(() => {
+        log("gateway is reachable");
+        reachable = true;
+      })
+      .catch(() => {
+        log("gateway is not reachable");
+      });
+    if (reachable) {
+      break;
+    }
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
+  }
+
+  if (reachable) {
+    axios.get(domain).then(res => {
+      log(res.status);
+      log(res.statusText);
+      log(res.data);
+      expect(res.status).toBe(200);
+      expect(res.statusText).toBe("OK");
+      expect(res.data).toContain("Directory listing for /");
+      expect(res.data).toContain("bin/");
+      expect(res.data).toContain("dev/");
+      expect(res.data).toContain("etc/");
+    });
+  } else {
+    throw new Error("Gateway is unreachable after multiple retries");
+  }
```

Comment:
You could also move this part to another function and use it to the other sections

Reply:
I think tests should be separate to make sure every test is correct and to be easier to maintain if any changes happen in one of the interfaces. What do you think?